storagestate/storagestate.go

```diff
@@ -280,17 +280,28 @@ func (storageState *StorageState) Remove(serviceID string) error {
 			continue
 		}
 
-		log.WithFields(log.Fields{
-			"instance":  stateStorageInfo.Instance,
-			"serviceID": stateStorageInfo.ServiceID,
-			"subjectID": stateStorageInfo.SubjectID,
-		}).Debug("Remove storage and state")
+		if err := storageState.removeServiceInstance(stateStorageInfo.InstanceIdent, stateStorageInfo.InstanceID); err != nil {
+			return err
+		}
+	}
 
-		if err := storageState.Cleanup(stateStorageInfo.InstanceIdent); err != nil {
-			return aoserrors.Wrap(err)
+	return nil
+}
+
+func (storageState *StorageState) RemoveServiceInstance(instanceIdent aostypes.InstanceIdent) error {
```

Comment:
Please write test on this function

Reply:
This function is tested in the existing test cases.
Would you like to add some new specific test cases?

Reply:
i'm ok if its used in some cases